src/fastpath/fastpath.c

```diff
@@ -254,7 +254,7 @@ void NORETURN fastpath_vm_fault(vm_fault_type_t type)
     }
 #endif
 
-#ifdef CONFIG_ARCH_RISCV64
+#ifdef CONFIG_ARCH_RISCV
     uint64_t addr;
```

Comment:
Is this is used for 32 and 64 bit now, using `word_t` might be better.

Reply:
Thanks, I've only been using RISCV64 so far and thought that would fix it for 32. The slowpath code which I based this off (handleVMFault in arch/riscv/kernel/vspace.c) seems to use uint64_t regardless of whether it is 32 or 64 bit. Any idea why this could be the case?

Reply:
It looks like this is wrong in the slowpath as well -- `read_stval` produces a `word_t`, and `seL4_Fault_VMFault_new` consumes a 32 bit value on rv32.

It's not catastrophic in this case, because `read_stval` will never produce values `seL4_Fault_VMFault_new` cannot consume, but it's better to fix the type.

Reply:
The RISC-V port has some places where the code could be cleaned up, because they seems just copy/paste to make it built somehow. And usually u64 often works for 32-bit also...